llvmcpy/_generator.py

```diff
@@ -23,38 +24,42 @@
 Classes = MutableMapping[str, List[Tuple[str, str, Any]]]
 EnumMap = MutableMapping[Union[int, str], Union[int, str]]
 
-ffi: Optional[FFI] = None
-
 unsigned_ints = set(("unsigned", "unsigned int", "unsigned long"))
 
-llvm_config: Optional[str] = None
-
-search_paths: str = ""
 
-version: str = ""
+class Context:
+    def __init__(self, llvm_config: Optional[str], search_paths: Optional[List[str]]):
```

Comment:
I'm not sure what was the point of providing both `search_paths` and `llvm_config` paths.

I'd suggest to have `__init__` just take a path to `llvm-config` (by default `which llvm-config`) and then add some factory methods to construct `Context` given a set of search paths and maybe another factory method that takes the LLVM version and looks up, e.g., `llvm-config-16` (or, as a fallback, just `llvm-config`) and checks the version is the one we expect.

What do you think?

Reply:
> I'm not sure what was the point of providing both search_paths and llvm_config paths.

They cover 2 different purposes (AFAIU):
* `llvm_config`: path to the `llvm-config` executable
* `search_paths`: paths to the auxiliary tools (e.g. gcc/clang)

> I'd suggest to have `__init__` just take a path to ...

See the second part of [this comment](https://github.com/revng/llvmcpy/pull/34#discussion_r1848458927)

Reply:
Dropped `search_paths`